src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHelpers.cs

```diff
@@ -113,11 +114,55 @@ internal static bool IsAllNullBytes(Span<byte> buffer)
             return true;
         }
 
-        // Returns a DateTimeOffset instance representing the number of seconds that have passed since the Unix Epoch.
-        internal static DateTimeOffset GetDateTimeFromSecondsSinceEpoch(double secondsSinceUnixEpoch)
+        // Converts the specified number of seconds that have passed since the Unix Epoch to a DateTimeOffset.
+        internal static DateTimeOffset GetDateTimeOffsetFromSecondsSinceEpoch(double secondsSinceUnixEpoch) =>
+            new DateTimeOffset((long)(secondsSinceUnixEpoch * TimeSpan.TicksPerSecond) + DateTime.UnixEpoch.Ticks, TimeSpan.Zero);
+
+        // Converts the specified DateTimeOffset to the number of seconds that have passed since the Unix Epoch.
+        internal static double GetSecondsSinceEpochFromDateTimeOffset(DateTimeOffset dateTimeOffset) =>
+            ((double)(dateTimeOffset.UtcDateTime - DateTime.UnixEpoch).Ticks) / TimeSpan.TicksPerSecond;
+
+        // If the specified fieldName is found in the provided dictionary and it is a valid double number, returns true and sets the value in 'dateTimeOffset'.
+        internal static bool TryGetDateTimeOffsetFromTimestampString(IReadOnlyDictionary<string, string> dict, string fieldName, out DateTimeOffset dateTimeOffset)
```

Comment:
I believe that literally every call to this passes an instance of `Dictionary<string, string>`, which means the parameter can be tightened to that type without a compile failure.

In this case, that's good because it'll call the non-virtual TryGetValue instead of the interface version (a virtual call), which faster at runtime.

It's the opposite guidance from public methods, where you want to be flexible in what you accept.

Reply:
Good tip.